### PR TITLE
Adding Email Validation To Form

### LINKED ISSUE #69: Simple email form validation
right now crap being typed into the email field leads to errors being logged in the backend. 

Let’s validate in the shared crate that only valid email address syntax is allowed or NO email at all. 

Use the same validation like the other existing validations on the front end and the backend

Use crate: https://docs.rs/email_address/latest/email_address/

### PR DESCRIPTION
**Pull request by Concelare**
This Pull Request Closes #69 

Adds Email Validation to Event Creation Form using the email_address crate, I also added another check of a `.` in the domain to make sure that the email isn't just `a@a` as that was originally passing using just the crate, as visible in the image below.
image

Final version is visible below.
image

Could possibly add a mailbox existance checker later on.

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
         Sentry.init({
             dsn: "https://2449cc52f1c847c6aa361dbde07e49c8@o4503948031426560.ingest.sentry.io/4503948034703360",
             environment: "local",
-            release: "0621722"
+            release: "6976421"
```
Comment (extrawurst) [inline frontend/index.html:33]: I usually don’t commit this one. It will always change and just clutter. Not found a good solution for this not to show up in git status

Comment (Concelare) [reply frontend/index.html:33]: Ah yea I didn't know if it needed to be updated but saw no issue if it was so just committed it, in future I'll make sure not to

File: backend/src/app.rs [hunk 2]
```diff
@@ -1082,6 +1083,34 @@ mod test {
...
+
+        let res = app
+            .create_event(AddEvent {
+                data: EventData {
+                    name: String::from("123456789"),
+                    description: String::from("123456789 123456789 123456789 !"),
+                    short_url: String::new(),
+                    long_url: None,
+                },
+                moderator_email: "a@a",
```
Comment (extrawurst) [inline backend/src/app.rs:1106]: lets also have one test that proofs a valid-non-empty email works

File: backend/src/app.rs [hunk 3]
```diff
@@ -1082,6 +1083,34 @@ mod test {
...
+            Arc::new(Payment::default()),
+            Tracking::default(),
+            String::new(),
+        );
+
+        let res = app
+            .create_event(AddEvent {
+                data: EventData {
+                    name: String::from("123456789"),
+                    description: String::from("123456789 123456789 123456789 !"),
```
Comment (extrawurst) [inline backend/src/app.rs:1102]: i think we have `TEST...` constants for this (in case we ever change the validation we only have to update these

File: backend/src/app.rs (hunk 2 above)
Comment (Concelare) [reply backend/src/app.rs:1106]: Will add now

File: backend/src/app.rs (hunk 3 above)
Comment (Concelare) [reply backend/src/app.rs:1102]: I can only see one for a TEST_VALID_QUESTION, I copied this template from the function below it

Comment (Concelare) [reply backend/src/app.rs]: Updated to use them now 👍

File: frontend/src/components/meta_popup.rs [hunk 4]
```diff
@@ -99,24 +90,33 @@ impl Component for MetaPopup {
...
-            Msg::ServerResponed(_) => {
+            Msg::ServerResponded(_) => {
                 self.send_pending = false;
                 ctx.props().on_close.emit(());
                 true
             }
         }
     }
 
+    fn changed(&mut self, ctx: &Context<Self>, _old_props: &Self::Properties) -> bool {
```
Comment (extrawurst) [inline frontend/src/components/meta_popup.rs:111]: not a fan of this function reordering, why did you move it below the `update`? it makes it hard to now see if the content of the function changed or just the place. so lets please reduce diffs in a PR to a minumum

Comment (Concelare) [reply frontend/src/components/meta_popup.rs]: I have rebased the changes to undo that change.

**Comment by extrawurst** [comment]
Great work! 🥳
